File: examples/node/nafath_server.js
/**
 * Minimal Nafath demo server (init + webhook)
 *
 * Run:
 *   npm i express
 *   node nafath_server.js
 *
 * Test locally:
 *   Expose http://localhost:3000/webhook/nafath via cloudflared/ngrok and
 *   set the public URL in the Authentica portal.
 *
 * Env:
 *   AUTHENTICA_API_KEY (required)
 *   BASE_URL (optional, defaults to https://api.authentica.sa)
 *   WEBHOOK_PASSWORD (shared secret used to verify webhook payloads)
 */

const express = require('express');

const app = express();
app.use(express.json());

const BASE_URL = process.env.BASE_URL || 'https://api.authentica.sa';
const API_KEY  = process.env.AUTHENTICA_API_KEY || 'YOUR_API_KEY';
const SECRET   = process.env.WEBHOOK_PASSWORD || 'your-secure-password';

// Initiate Nafath verification for a user
app.post('/nafath/init', async (req, res) => {
    try {
        const r = await fetch(`${BASE_URL}/api/v2/verify-by-nafath`, {
            method: 'POST',
            headers: { 'Accept': 'application/json','Content-Type': 'application/json','X-Authorization': API_KEY },
            body: JSON.stringify({ user_id: (req.body && req.body.user_id) || 'demo-user' })
        });
        const j = await r.json().catch(() => ({}));
        res.status(r.status).json(j);
    } catch (e) { res.status(500).json({ error: e.message }); }
});

// Receive webhook from Authentica after user acts in Nafath app
app.post('/webhook/nafath', (req, res) => {
    if (!req.body || req.body.Password !== SECRET) return res.sendStatus(401);
    console.log('Nafath result:', { status: req.body.Status, nationalId: req.body.NationalId });
    res.sendStatus(200);
});

app.listen(3000, () => console.log('Nafath demo server on http://localhost:3000'));

File: examples/python/nafath_server.py
"""
Minimal Nafath demo server (init + webhook)

Run:
  pip install fastapi uvicorn requests
  uvicorn nafath_server:app --port 3000

Local testing:
  Expose http://localhost:3000/webhook/nafath via cloudflared/ngrok and set the public URL in the portal.

Env:
  AUTHENTICA_API_KEY (required)
  BASE_URL (optional, defaults to https://api.authentica.sa)
  WEBHOOK_PASSWORD (shared secret used to verify webhook payloads)
"""

import os, json, requests
from fastapi import FastAPI, Request, HTTPException

app = FastAPI()
BASE_URL = os.getenv('BASE_URL','https://api.authentica.sa')
API_KEY  = os.getenv('AUTHENTICA_API_KEY','YOUR_API_KEY')
SECRET   = os.getenv('WEBHOOK_PASSWORD','your-secure-password')

@app.post('/nafath/init')
def nafath_init(payload: dict):
    r = requests.post(
        f"{BASE_URL}/api/v2/verify-by-nafath",
        headers={'Accept':'application/json','Content-Type':'application/json','X-Authorization':API_KEY},
        data=json.dumps({'user_id': payload.get('user_id','demo-user')})
    )
    return {'status': r.status_code, 'json': r.json()}

@app.post('/webhook/nafath')
async def nafath_webhook(req: Request):
    data = await req.json()
    if data.get('Password') != SECRET:
        raise HTTPException(status_code=401, detail='Invalid secret')
    print('Nafath result:', {'status': data.get('Status'), 'nationalId': data.get('NationalId')})
    return {'ok': True}
